### ISSUE TITLE
[HELP] image png within full-screen undecorated window still not expanding to entire screen real-estate

### ISSUE THREAD

**Issue by omac777** [gtk-rs/gtk4-rs#1037]
My monitor is width 1920 and height 1080.
My image png is also width 1920 and height 1080.
I have set the main window to be undecorated and full screen(1920x1080).
The image rendered seems to be much smaller than the full screen, but when I check the image info, it's correct:(1920x1080).

Does anybody have any advice to isolate why the image png isn't occupying the entire full screen?
Thank you in advance.

```
use gtk4::prelude::*;
use gtk4::Application;
use gtk4::ApplicationWindow;
use gtk4::builders::FlowBoxBuilder;
use gtk4::Button;
use gtk4::Image;
use gdk_pixbuf::Pixbuf;
use gtk4::Align;

fn main() {
    let application = gtk4::Application::new(
        Some("com.mfw.sgi_viewer"),
        Default::default(),
    );
    application.connect_activate(build_ui);
    application.run();
}

fn build_ui(app: &gtk4::Application) {
    let (thePixbufFormat, theWidth, theHeight) = Pixbuf::file_info("/home/davidm/photoqt/my-gtk-app/final-0001.png").unwrap();
    let window = gtk4::ApplicationWindow::builder()
        .default_width(theWidth)
        .default_height(theHeight)
        .application(app)
        .title("sgi_viewer")
	.decorated(false)
	.fullscreened(true)
        .build();

    let pixbuf = Pixbuf::from_file("/home/davidm/photoqt/my-gtk-app/final-0001.png").unwrap();
    let (thePixbufFormat, theWidth, theHeight) = Pixbuf::file_info("/home/davidm/photoqt/my-gtk-app/final-0001.png").unwrap();
    println!("thePixbufFormat:<<{:?}>>, theWidth:<<{:?}>>, theHeight:<<{:?}>>", thePixbufFormat, theWidth, theHeight);
    let _image = Image::from_pixbuf(Some(&pixbuf));

    _image.set_size_request(theWidth, theHeight);
    //_image.set_expand(true);
    _image.set_hexpand(true);
    _image.set_vexpand(true);
    _image.set_valign(Align::Fill);
    _image.set_halign(Align::Fill);
    window.set_child(Some(&_image));
    window.show();
}
```

**Comment by omac777** [comment]
I took a screenshot and measured it in gimp.
The full screen is as stated 1920x1080.
The image is landing in a rect of 1080x600 centered in the screenshot.  I didn't tell it to center the image within the window anywhere.

**Comment by jf2048** [comment]
`Image` is not meant for this, use a `gtk4::Picture` instead.

**Comment by bilelmoussaoui** [comment]
Yes, you need to use gtk::Picture here. Image is for loading icons

**Comment by omac777** [comment]
Ultimately I resorted to using Pixbuf and holding into a drawingarea, and painting there.

    On Friday, May 20, 2022, 05:10:52 a.m. EDT, Bilal Elmoussaoui ***@***.***> wrote:  
 
 


Closed #1037 as completed.
